components/services/ipfs/ipfs_service_impl.cc

```diff
@@ -108,22 +109,62 @@ bool LaunchProcessAndExit(const base::FilePath& path,
   return true;
 }
 
+void MonitorChild(base::ProcessHandle p_handle,
```

Comment:
I will fix the duplication by wrapping a `ChildMonitor` class and also global `sigaction` calls don't matter because `IpfsServiceImpl` and `TorLauncherImpl` are in different utility processes.

Reply:
fixed in https://github.com/brave/brave-core/pull/8620/commits/7757acb9179b6e2c366625d8a61168bd0f106746